### PR TITLE
Add span normalise

### PR DESCRIPTION
**Pull request by jeromekelleher**
Tick off a few issues with the divmat implementation.

### CODE REVIEW

File: python/tskit/trees.py [hunk 1]
```diff
@@ -7852,27 +7856,42 @@ def worker(sub_windows):
...
-        windows = [0, self.sequence_length] if windows is None else windows
-
+        windows = self.parse_windows(windows)
         mode = "site" if mode is None else mode
 
-        # NOTE: maybe we want to use a different default for num_threads here, just
-        # following the approach in GNN
+        # FIXME this logic should be merged into __run_windowed_stat if
+        # we generalise the num_threads argument to all stats.
         if num_threads <= 0:
```
Comment (petrelharp) [inline python/tskit/trees.py:7873]: not `<= 1`?

**Review by petrelharp — APPROVED** [review]
LGTM!

File: python/tskit/trees.py (hunk 1 above)
Comment (jeromekelleher) [reply python/tskit/trees.py:7873]: Subtle difference - you might want to do the work in *one* background worker thread. Zero or less means that the work without threads, but default (which simplifies debugging a lot)

**Comment by jeromekelleher** [comment]
Codecov is just wrong about the coverage here, ignoring.

**Pull request merged**